Feat: Segmentation methods for generating names and finding files

MattBuckley-TfN opened the pull request:
Describe Changes
---
- Added method to generate name from segment parameters e.g. "p1_m3". 
- Added method to generate segment tuples e.g. (1, 3).
- Added method to find files in folder based on names.
- Added basic tests for the new methods.
- Added isort action

Task Checklist
----
- [x] Have unittests been added (testing individual functions and their edge cases) - yes
- [x] Have integration tests been added (if applicable, to test modules of functionality) - no
- [x] Have new dependencies been added? - no
  - [x] Have they been added to either `requirements.txt` or `requirements_dev.txt`. - NA
  - [x] Have they been added to `setup.cfg` - NA

MattBuckley-TfN commented:
Not yet ready for review

isaac-tfn commented:
> @isaac-tfn - not yet ready for review

Will take a look when ready - is this just more descriptive naming for subset segments?

MattBuckley-TfN commented:
> Will take a look when ready - is this just more descriptive naming for subset segments?

It produces segment names ready for files e.g. "p1_m3_ca1_tp2", I'm also adding a function that will generate full filenames, its mainly useful when working with matrices but could be used wherever else segmentation is split across different files. Thinking about it this could be included in caf.mat itself rather than as a method on segmentation.

MattBuckley-TfN commented:
I've added isort action and fixed some linting errors. I haven't fixed the mypy linting errors, which are in `data_structures` because I don't fully understand the reason for the updating path name (below).

```python
out_path = Path(out_path)
if "." not in out_path.name:
    out_path.name = out_path.name
```

@isaac-tfn - I think this should be ready for review and the mypy issues can be dealt with at a later time

isaac-tfn reviewed (approved):
This all looks good to me!
